src/openrct2/diagnostic.c

```diff
@@ -30,6 +31,15 @@ const char * _level_strings[] = {
 	"INFO"
 };
 
+// returns the current system time in a printable format                                                                                                                                                    
+char* getTime() {
+        time_t rawTime;
+        time(&rawTime);
+        static char curTime[128];
+        strftime(curTime, sizeof(curTime), "[%FT%T]", localtime(&rawTime));
```

Comment:
I wrote a new function that uses these functions to fill rct2_date and rct2_time objects and prints their contents.